### ISSUE TITLE
i get different results than expected

### ISSUE THREAD

**Issue by wendysalonika** [mesnilgr/is13#25]
Hi,
I'm running your code rnnslu.py for ATIS (the the 5 folds of data you provide us with)
(DeepLearningTutorials/code/rnnslu.py)

I expected to get F1=94.98% as mentioned in the paper:
'Using Recurrent Neural Networks for Slot Filling in Spoken Language Understanding'
(table 3: Elman/Single,context(9))

Instead, the best performance with this code is F1=94.058% for cont_win=9 and emb_dim=100.
Is that what I'm suppose to get?

Thank you in advance!

**Comment by mesnilgr** [comment]
Hi,

I ran experiments in the paper with different hyper-parameters than the ones specified in the code.
The problem is I don't have access to my university account (where all my experiment results stored) since I graduated. Let me check with them if I can get guest access or not.

Otherwise, the way I proceed was to launch 200 experiments with random search for hyper-parameters and take the best on valid F1. It's specified in the paper section VI-C. Another thing, there is a lot of variance in the performance as you can see in Table 2, so F1=94.058% looks fine to me.

Best,
Grégoire

**Comment by wendysalonika** [comment]
so I guess the key is to run the experiments with random sampling of the hyper-parameters.

I've already tested d ∈ {3, 5, ... , 17} , the embedding dimension, d e ∈ {50, 100}
but as for the depth, H ∈ {1,2}--> does depth 2 mean to stack another RNN top of the other?
(as done in Deep LSTM?)
thnx

**Comment by mesnilgr** [comment]
the depth did not bring any improvement in my experiments so you can keep H=1. (depth=2 means I tried stacking a second layer as done in deep LSTM you' re right).

I also played with the learning rate and the random seed for both initializing the parameters and shuffling the examples at each epoch.

Let me know the outcome of your experiments to make sure you are able to reproduce the results.
Best, 
Grégoire

**Comment by wendysalonika** [comment]
I appreciate your reply
it's really helful

**Comment by wendysalonika** [comment]
I hope I'm not being too annoying owing to my questions...
but I still can't get the highest F1 of the paper with only 200 experiments (for rnnslu.py)
--->how many different 'lr' did you test? (should I check, let's say, about 20? )   and 
--->the 3 seeds u used where 2,3,4 ? (as suggested in Random Search for Hyper-Parameter Optimization??)
thank you

**Comment by mesnilgr** [comment]
I still don't have access to the hyper parameters.
From my memory, seed was 123, 456, 789.
Also the train/valid split varies among 0..4 (see the download section in
the readme that I updated recently).
The lr is sampled from a uniform between 0.01 and 0.1, so there is 200
different learning rates if you launch 200 experiments.
I hope this helps.

Le mardi 8 mars 2016, wendysalonika notifications@github.com a écrit :

> I hope I'm not being too annoying owing to my questions...
> but I still can't get the highest F1 of the paper with only 200
> experiments (for rnnslu.py)
> --->how many different 'lr' did you test? (should I check, let's say,
> about 20? ) and
> --->the 3 seeds u used where 2,3,4 ? (as suggested in Random Search for
> Hyper-Parameter Optimization??)
> thank you
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/mesnilgr/is13/issues/25#issuecomment-193653591.

## 

Grégoire

**Comment by wendysalonika** [comment]
hey there,
I know my question is stupid (so sorry beforehand) i just want to make sure i don't do sth wrong

when you use 5-fold cross-validation, the test F1 score presented on your paper is the mean F1 of the 5 folds?? right? (not the highest F1 of the 5 folds)

thnxxxxxx

**Comment by phoenixia** [comment]
Hey,

No such thing as a stupid question.

I did something different because I was using random search for hyper-parameter selection and I considered the split as a hyper-parameter (which is kind of counter intuitive but fitted well in my framework).

I trained on 5 different splits, pick the the model that got the best validation F1 no matter the split, then reported the test F1 in the paper (the test set is the same since the split only affects the train and valid set).

To answer your question, I did no 5-fold cross validation because it tends to be time consuming and I wanted to explore many different research avenues in a single paper.

Cheers

Gregoire

**Comment by wendysalonika** [comment]
thank you so much for your reply,
in the last reply you mention sth about training 5 different splits - are these splits the 5 folds of the data? or you read the whole dataset and you split it on your own?
(did you use atisfold() or atisfull() to read the data?)

what I understand is that:
- train fold_0 ==> validation_F1 = v0 ,  test_F1 = t0
- train fold_1 ==> validation_F1 = v1 ,  test_F1 = t1
  ...
- train fold_4 ==> validation_F1 = v4 ,  test_F1 = t4

and you reported the test_F1 for the highest validation_F1. right??
that's what you did in the paper 'Using Recurrent Neural Networks for Slot Filling in Spoken Language Understanding'??

I really appreciate all your help

**Comment by mesnilgr** [comment]
_are these splits the 5 folds of the data? or you read the whole dataset and you split it on your own?
(did you use atisfold() or atisfull() to read the data?)_

**I use atis fold and so you can download exactly the same splits.**

_and you reported the test_F1 for the highest validation_F1. right??
that's what you did in the paper 'Using Recurrent Neural Networks for Slot Filling in Spoken Language Understanding'??_

**Yes**

**Comment by wendysalonika** [comment]
thnx
